Error building with SQL (SOCI) database back-end support.

This error occurs when trying to build with libsoci-dev package (3.2.3-1ubuntu2) installed on Ubuntu 16.04 

```
In file included from /usr/include/soci/soci.h:33:0,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/rowset.h:157:16: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
     const std::auto_ptr<statement> st_;
                ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /home/littlecat/Astron/src/util/Datagram.h:9,
                 from /home/littlecat/Astron/src/util/DatagramIterator.h:2,
                 from /home/littlecat/Astron/src/database/DBOperation.h:7,
                 from /home/littlecat/Astron/src/database/DatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/OldDatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/soci/soci.h:33:0,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/rowset.h:158:16: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
     const std::auto_ptr<T> define_;
                ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /home/littlecat/Astron/src/util/Datagram.h:9,
                 from /home/littlecat/Astron/src/util/DatagramIterator.h:2,
                 from /home/littlecat/Astron/src/database/DBOperation.h:7,
                 from /home/littlecat/Astron/src/database/DatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/OldDatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/soci/soci.h:34:0,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/session.h:43:41: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
     void set_query_transformation_(std::auto_ptr<details::query_transformation_
                                         ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /home/littlecat/Astron/src/util/Datagram.h:9,
                 from /home/littlecat/Astron/src/util/DatagramIterator.h:2,
                 from /home/littlecat/Astron/src/database/DBOperation.h:7,
                 from /home/littlecat/Astron/src/database/DatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/OldDatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/soci/soci.h:34:0,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/session.h: In member function ‘void soci::session::set_query_transformation(T)’:
/usr/include/soci/session.h:80:14: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
         std::auto_ptr<details::query_transformation_function> qtf(new details::
              ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /home/littlecat/Astron/src/util/Datagram.h:9,
                 from /home/littlecat/Astron/src/util/DatagramIterator.h:2,
                 from /home/littlecat/Astron/src/database/DBOperation.h:7,
                 from /home/littlecat/Astron/src/database/DatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/OldDatabaseBackend.h:2,
                 from /home/littlecat/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
/home/littlecat/Astron/src/database/SociSQLDatabase.cpp: In constructor ‘SociSQLDatabase::SociSQLDatabase(ConfigNode, doid_t, doid_t)’:
/home/littlecat/Astron/src/database/SociSQLDatabase.cpp:46:22: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
         if(col_index != string::npos && col_index > sqr_index) {
                      ^
/home/littlecat/Astron/src/database/SociSQLDatabase.cpp: In member function ‘doid_t SociSQLDatabase::pop_next_id()’:
/home/littlecat/Astron/src/database/SociSQLDatabase.cpp:557:53: error: ‘BOOST_BINARY’ was not declared in this scope
         if(!(first.bounds().bits() & BOOST_BINARY(10))) {
                                                     ^
cc1plus: all warnings being treated as errors
CMakeFiles/astrond.dir/build.make:998: recipe for target 'CMakeFiles/astrond.dir/src/database/SociSQLDatabase.cpp.o' failed
make[2]: *** [CMakeFiles/astrond.dir/src/database/SociSQLDatabase.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/astrond.dir/all' failed
make[1]: *** [CMakeFiles/astrond.dir/all] Error 2
Makefile:94: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/src/database/SociSQLDatabase.cpp
+++ b/src/database/SociSQLDatabase.cpp
@@ -41,8 +41,8 @@ class SociSQLDatabase : public OldDatabaseBackend
         m_log = new LogCategory(m_backend, log_name.str());
 
         string server = database_address.get_rval(dbeconfig);
-        int col_index = server.find_last_of(":");
-        int sqr_index = server.find_last_of("]");
+        size_t col_index = server.find_last_of(":");
+        size_t sqr_index = server.find_last_of("]");
         if(col_index != string::npos && col_index > sqr_index) {
             m_db_host = server.substr(0, col_index);
             m_db_port = stoi(server.substr(col_index + 1));
@@ -554,7 +554,7 @@ class SociSQLDatabase : public OldDatabaseBackend
         // Get next available id from m_free_ids set
         interval_t first = *m_free_ids.begin();
         doid_t id = first.lower();
-        if(!(first.bounds().bits() & BOOST_BINARY(10))) {
+        if(!(first.bounds().bits() & 2)) {
             id += 1;
         }
 

